validation/distributed_simulations/distributed_geostrophic_adjustment.jl

```diff
@@ -66,26 +71,20 @@ gravity_wave_speed = sqrt(g * grid.Lz) # hydrostatic (shallow water) gravity wav
 Δt = 2 * model.grid.Δxᶜᵃᵃ / gravity_wave_speed
 simulation = Simulation(model; Δt, stop_iteration = 1000)
 
-ut = []
-vt = []
-ηt = []
-
-save_u(sim) = push!(ut, deepcopy(sim.model.velocities.u))  
-save_v(sim) = push!(vt, deepcopy(sim.model.velocities.v))
-save_η(sim) = push!(ηt, deepcopy(sim.model.free_surface.η))
-
 function progress_message(sim) 
     @info @sprintf("[%.2f%%], iteration: %d, time: %.3f, max|w|: %.2e",
     100 * sim.model.clock.time / sim.stop_time, sim.model.clock.iteration,
     sim.model.clock.time, maximum(abs, sim.model.velocities.u))
 end
 
-simulation.callbacks[:save_η]   = Callback(save_η, IterationInterval(1))
-simulation.callbacks[:save_v]   = Callback(save_v, IterationInterval(1))
-simulation.callbacks[:save_u]   = Callback(save_u, IterationInterval(1))
 simulation.callbacks[:progress] = Callback(progress_message, IterationInterval(10))
 
-run!(simulation)
 
-jldsave("variables_rank$(rank).jld2", v=vt, η=ηt, u=ut)
+outputs = merge(model.velocities, model.free_surface)
```

Comment:
I think the issue it with the output writer, this should fix it.

```suggestion
outputs = merge(model.velocities, model.free_surface.η)
```